### PR TITLE
Update Libp2p to v0.15.1

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Dependency Update

**What does this PR do? Why is it needed?**

- [x] Updates `go-libp2p` to v0.15.1 and updates all its related transitive dependencies along with it.

**Which issues(s) does this PR fix?**

This PR is split out from #9772 to make it smaller and more manageable to merge in separately.

**Other notes for review**

### CODE REVIEW

File: beacon-chain/p2p/discovery.go [hunk 1]
```diff
@@ -454,11 +453,7 @@ func peersFromStringAddrs(addrs []string) ([]ma.Multiaddr, error) {
 }
 
 func multiAddrFromString(address string) (ma.Multiaddr, error) {
-	addr, err := iaddr.ParseString(address)
-	if err != nil {
-		return nil, err
-	}
-	return addr.Multiaddr(), nil
+	return ma.NewMultiaddr(address)
```
Comment (potuz) [inline beacon-chain/p2p/discovery.go:456]: What's up with this, my not change then calls to `multiAddrFromString`?

Comment (prestonvanloon) [reply beacon-chain/p2p/discovery.go:456]: github.com/ipfs/go-ipfs-addr is deprecated and no longer works with new updates to `github.com/multiformats/go-multiaddr`

Comment (prestonvanloon) [reply beacon-chain/p2p/discovery.go:456]: I see what you are saying as feedback though. There is no longer a need to have this helper function.

Let's follow up refactor this in another PR to minimize changes.

**Review by prestonvanloon — APPROVED** [review]
Thanks!

Please note that this PR also updates go-ethereum to https://github.com/ethereum/go-ethereum/commit/b97f57882c14.

This was a necessary update for #9772 and the update is not technically required for libp2p, but it seems safe to update anyway.

**Pull request merged**